test/integration/Makefile

```diff
@@ -25,154 +25,138 @@ EUID := $(shell id -u -r)
 
-includes: setup
-	ansible-playbook test_includes.yml -i $(INVENTORY) -e outputdir=$(TEST_DIR) -e @$(VARS_FILE) $(CREDENTIALS_ARG) $(TEST_FLAGS)
-
-pull:	pull_run pull_no_127 pull_limit_inventory
+includes:
+	(cd targets/includes && ./runme.sh $(TEST_FLAGS))
 
 pull_run:
-	ansible-pull -d $(MYTMPDIR) -U https://github.com/ansible-test-robinro/pull-integration-test.git $(TEST_FLAGS) | grep MAGICKEYWORD; \
-    RC=$$? ; \
-	rm -rf $(MYTMPDIR); \
-    exit $$RC
+	(cd targets/pull_run && ./runme.sh $(TEST_FLAGS))
 
-# test for https://github.com/ansible/ansible/issues/13681
 pull_no_127:
-	ansible-pull -d $(MYTMPDIR) -U https://github.com/ansible-test-robinro/pull-integration-test.git $(TEST_FLAGS) | grep -v 127\.0\.0\.1; \
-    RC=$$? ; \
-	rm -rf $(MYTMPDIR); \
-    exit $$RC
+	(cd targets/pull_no_127 && ./runme.sh $(TEST_FLAGS))
 
-# test for https://github.com/ansible/ansible/issues/13688
 pull_limit_inventory:
-	ansible-pull -d $(MYTMPDIR) -U https://github.com/ansible-test-robinro/pull-integration-test.git $(TEST_FLAGS); \
-    RC=$$? ; \
-	rm -rf $(MYTMPDIR); \
-    exit $$RC
-
+	(cd targets/pull_limit_inventory && ./runme.sh $(TEST_FLAGS))
 
-unicode: setup
-	ansible-playbook unicode.yml -i $(INVENTORY) -e outputdir=$(TEST_DIR) -e @$(VARS_FILE) -v $(TEST_FLAGS) -e 'extra_var=café'
-	# Test the start-at-task flag #9571
-	ansible-playbook unicode.yml -i $(INVENTORY) -e outputdir=$(TEST_DIR) -e @$(VARS_FILE) -v --start-at-task '*¶' -e 'start_at_task=True' $(TEST_FLAGS)
+unicode:
+	(cd targets/unicode && ./runme.sh $(TEST_FLAGS))
 
-test_templating_settings: setup
-	ansible-playbook test_templating_settings.yml -i $(INVENTORY) -e outputdir=$(TEST_DIR) -e @$(VARS_FILE) $(CREDENTIALS_ARG) -v $(TEST_FLAGS)
+test_templating_settings:
+	(cd targets/templating_settings && ./runme.sh $(TEST_FLAGS))
 
-test_gathering_facts: setup
-	ansible-playbook test_gathering_facts.yml -i $(INVENTORY) -e outputdir=$(TEST_DIR) -e @$(VARS_FILE) -v $(TEST_FLAGS)
+test_gathering_facts:
+	(cd targets/gathering_facts && ./runme.sh $(TEST_FLAGS))
 
-environment: setup
-	ansible-playbook test_environment.yml -i $(INVENTORY) -e outputdir=$(TEST_DIR) -e @$(VARS_FILE) $(CREDENTIALS_ARG) $(TEST_FLAGS)
+environment:
+	(cd targets/environment && ./runme.sh $(TEST_FLAGS))
 
 non_destructive: setup
 	ansible-playbook non_destructive.yml -i $(INVENTORY) -e outputdir=$(TEST_DIR) -e @$(VARS_FILE) $(CREDENTIALS_ARG) -v $(TEST_FLAGS)
 
-# For our Docker images, which identify themselves with "ENV container=docker", automatically run the test_async target.
-# Otherwise, skip it, since we don't know if local ssh is available. You can always run the test_async target manually.
-ifeq ($(container),docker)
-test_async_conditional: test_async
-else
-test_async_conditional:
-endif
+# Tests which can be run when running as root and a local SSH daemon is available.
+test_root_ssh: test_connection_local test_connection_chroot test_connection_ssh test_connection_paramiko_ssh test_delegate_to
```

Comment:
Is the plan that these targets will be called directly from the new script?
I notice that you've removed `test_connection` from `other:`